channel/channel.go

```diff
@@ -8,15 +8,21 @@ import (
 
 const defaultChannelTimeout = time.Second * 1
 
-var BlockedPush = string("Push blocked on channel: ")
-var UnblockedPush = string("Push unblocked on channel: ")
-var BlockedPop = string("Pop blocked on channel: ")
-var UnblockedPop = string("Pop unblocked on channel: ")
-
 type Timed[T any] struct {
 	c chan T
 }
 
+type Error string
+
+const (
+	ErrFailedToSend    = Error("Could not send data on channel.")
+	ErrFailedToReceive = Error("Could not receive data from channel")
+)
+
+func (e Error) Error() string {
+	return string(e)
+}
+
 // WithExpiration creates a new channel of the given size and type
 func WithExpiration[T any](bufSize int) Timed[T] {
 	Logger = Logger.With().Int("size", bufSize).Logger()
```

Comment:
This seems very buggy. We're overwriting the logger every time a new channel is created.
We should put a new logger (sub-)instance in the `Timed[T]` structure and use it from there.
I.e., instead of calling `Logger.*` in all the functions, we could call `c.log.*`